cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -439,3 +482,40 @@ func (c *CStorPoolController) getDeviceIDs(csp *apis.CStorPool) ([]string, error
 	// TODO: Add error handling
 	return pool.GetDeviceIDs(csp)
 }
+
+func (c *CStorPoolController) getBlockDeviceList(csp *apis.CStorPool, namespace string) (*blockdevice.BlockDeviceList, error) {
+	bdNames, err := pool.GetBlockDeviceNames(csp)
+	if err != nil {
+		return nil, err
+	}
+	customBDList := &blockdevice.BlockDeviceList{
+		BlockDeviceList: &ndmapis.BlockDeviceList{},
+	}
+	for _, bdName := range bdNames {
+		bdObj, err := c.ndmClientset.OpenebsV1alpha1().BlockDevices(namespace).Get(bdName, metav1.GetOptions{})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get blockdevice")
+		}
+		customBDList.BlockDeviceList.Items = append(customBDList.BlockDeviceList.Items, *bdObj)
+	}
+	return customBDList, nil
+}
+
+func validateBlockDeviceClaimStatus(customBDList *blockdevice.BlockDeviceList, bdcListBuilder *bdc.ListBuilder) error {
+	for _, bdObj := range customBDList.BlockDeviceList.Items {
+		bdObj := bdObj
+		customBDObj := &blockdevice.BlockDevice{
+			BlockDevice: &bdObj,
+		}
+		if customBDObj.IsClaimed() {
```

Comment:
are you following v1alpha1 for BD?

Reply:
why v1alpha1?

Reply:
Since we are following v1alpha1 in cstor-operator I am following same thing here